src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+    * @param selectedJoints Seq of Joints that have been identified as seed joints
+    * @param tolerance Acceptable deviation from ideal volume per processor
+    * @param numProcessors Number of processors that will be used in the analysis
+    * @return
+    */
+  @tailrec
+  private def cycleJointSet(jointSet: Seq[Joint], initialBlock: Block, volumePerProcessor: Double,
+                            totalVolume: Double, selectedJoints: Seq[Joint], tolerance: Double,
+                            numProcessors: Int): Seq[Joint] = {
+    if (selectedJoints.length == numProcessors - 1) {
+      // Enough seed joints have been found
+      selectedJoints
+    } else if (jointSet.length > 1) {
+      // More than 1 joint in input joint set
+      val jointOption = testVolumes(jointSet.head, jointSet.tail.head, initialBlock,
+        volumePerProcessor, totalVolume, tolerance)
+
+      if (jointOption.isEmpty) {
+        // Joint did not meet criteria
+        cycleJointSet(jointSet.tail, initialBlock, volumePerProcessor, totalVolume, selectedJoints,
+          tolerance, numProcessors)
+      } else {
+        // Joint meets criteria
+        val (seedJoint, remainingVolume) = jointOption.get
+        if (remainingVolume.volume <= volumePerProcessor) {
+          // Remaining volume small enough, return
+          seedJoint +: selectedJoints
+        } else {
+          // Keep cycling through joints
+          cycleJointSet(jointSet.tail, remainingVolume, volumePerProcessor, totalVolume,
+            seedJoint +: selectedJoints, tolerance, numProcessors)
+        }
+      }
+    } else {
+      // Only one joint in input joint set
+      val lastBlocks = initialBlock cut jointSet.head
+      val nonRedundantLastBlocks = lastBlocks map { case block @ Block(blockCenter, _, _) =>
+          Block(blockCenter, block.nonRedundantFaces)
+      }
+      val sortedLastBlocks = nonRedundantLastBlocks.sortWith(_.volume < _.volume)
+      if (sortedLastBlocks.length == 1) {
+        // Joint does not intersect volume
+        selectedJoints
+      } else {
+        // Joint intersect volume
+        jointSet.head +: selectedJoints
+      }
+    }
+  }
+
+  /**
+    * Test which, if any, of two input joints divides the input block into blocks of
+    * desired volume.
+    *
+    * @param joint1 First input joint
+    * @param joint2 Second input joint
+    * @param inputBlock Block that is to be subdivided
+    * @param desiredVolume Desired volume of subdivision(s)
+    * @param totalVolume Total volume of rock volume in analysis
+    * @param tolerance Acceptable deviation from ideal volume per processor
+    * @return Returns tuple containing joint that provides best division of volume and remaining volume that is to be
+    *         subdivided further. If neither joint yields acceptable subdivision, function returns None.
+    */
+  private def testVolumes(joint1: Joint, joint2: Joint, inputBlock: Block,
```

Comment:
I have to admit I found this function pretty confusing.

Why do we consider joints two at a time instead of one at a time? May want to add some comments explaining this.